### ISSUE TITLE
Bug: [no-floating-promises] doesn't detect nested arrays to `Promise.all()`

### ISSUE THREAD

**Issue by libre-man** [typescript-eslint/typescript-eslint#11257]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.8.2&fileType=.tsx&code=IYZwngdgxgBAZgV2gFwJYHsI2ACgJQwDeAUDNgO7CrIwAKATugLaogCmAdMADbc4DapGPwbNWnemxDpuANzb4AugBohivMQC%2BxIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUFs%2B5AAgC4CeADigMbQCWphAtCvNQHaED0LA9vQGbxcAhoVYBzeqWhcAttWQp0URNCnRI4MAF8QmoA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
async function a() {
  await Promise.all([
    [Promise.resolve()],
  ])
}
```

### ESLint Config

```javascript
{
  "rules": {
            "@typescript-eslint/no-floating-promises": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

I would expect this code to be flagged, as `Promise.all([[promise]])` doesn't await `promise`.

### Actual Result

There was no issue detected.

### Additional Info

This is quite a subtle bug and it would be great if `typescript-eslint` could catch it.

**Comment by kirkwaiblinger** [comment]
Hi @libre-man!

Note that https://github.com/typescript-eslint/typescript-eslint/issues/1804 is related and also solves this specific case, by different logic.

The trouble with detecting this case in `no-floating-promises` is that we're only looking for statements whose type is `Promise<T>` or (since #7897) `Promise<T>[]`. The statement in this issue has type `Promise<T>[][]`... While that's clearly an error in this particular case, I'm not sure we can generalize that (and if so, what's an appropriate bound to this generalization - e.g. what about `Promise<T>[][][]`?). Unless there's a compelling answer to that, I'm tentatively leaning towards cleaning this in favor of #1804. WDYT?

**Comment by bradzacher** [comment]
I would agree that #1804 would cover this.

**Comment by libre-man** [comment]
I guess indeed that #1804 is basically the same ticket indeed. Closing it in favor of that one seems fine.
